packages/catalog-client/src/CatalogClient.ts

```diff
@@ -35,55 +36,30 @@ export class CatalogClient implements CatalogApi {
     this.discoveryApi = options.discoveryApi;
   }
 
-  private async getRequired(path: string): Promise<any> {
-    const url = `${await this.discoveryApi.getBaseUrl('catalog')}${path}`;
-    const response = await fetch(url);
-
-    if (!response.ok) {
-      const payload = await response.text();
-      const message = `Request failed with ${response.status} ${response.statusText}, ${payload}`;
-      throw new Error(message);
-    }
-
-    return await response.json();
+  async getLocationById(id: String): Promise<Location | undefined> {
+    return await this.getOptional(`/locations/${id}`);
   }
 
-  private async getOptional(path: string): Promise<any | undefined> {
-    const url = `${await this.discoveryApi.getBaseUrl('catalog')}${path}`;
-    const response = await fetch(url);
+  async getEntities(request?: CatalogEntitiesRequest): Promise<Entity[]> {
```

Comment:
I don't think it would be too early to make this `{items: Entity[]}` at all tbh

Reply:
done